packages/patternfly-4/react-core/src/components/OptionsMenu/examples/OptionsMenu.md

```diff
@@ -457,3 +459,63 @@ class PlainWithText extends React.Component {
   }
 }
 ```
+
+```js title=Grouped-items-with-titles
+import React from 'react';
+import { OptionsMenu, OptionsMenuItem, OptionsMenuToggle, OptionsMenuGroup } from '@patternfly/react-core';
+
+class GroupedItems extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: false,
+      toggleTemplateText: "Options menu",
+      selectedOption: "groupOption1"
+    };
+    
+    this.onToggle = () => {
+      this.setState({
+        isOpen: !this.state.isOpen
+      });
+    };
+    
+    this.onSelect = event => {
+      const id = event.currentTarget.id;
+      this.setState(() => {
+        return { selectedOption: id };
+      });
+    };
+  }
+  
+  render() {
+    const { isOpen, selectedOption, toggleTemplateText } = this.state;
+    
+    const menuGroups = [
+      <OptionsMenuGroup hasSeparator key="group0">
+        <OptionsMenuItem onSelect={this.onSelect} isSelected={selectedOption === "groupOption1"} id="groupOption1" key="option 1">Option 1</OptionsMenuItem>
+        <OptionsMenuItem onSelect={this.onSelect} isSelected={selectedOption === "groupOption2"} id="groupOption2" key="option 2">Option 2</OptionsMenuItem>
+      </OptionsMenuGroup>,
+      <OptionsMenuGroup groupTitle="Group 1" hasSeparator key="group1">
+        <OptionsMenuItem onSelect={this.onSelect} isSelected={selectedOption === "groupOption3"} id="groupOption3" key="option 3">Option 1</OptionsMenuItem>
+        <OptionsMenuItem onSelect={this.onSelect} isSelected={selectedOption === "groupOption4"} id="groupOption4" key="option 4">Option 2</OptionsMenuItem>
+      </OptionsMenuGroup>,
+      <OptionsMenuGroup groupTitle="Group 2" key="group2">
+        <OptionsMenuItem onSelect={this.onSelect} isSelected={selectedOption === "groupOption5"} id="groupOption5" key="option 5">Option 1</OptionsMenuItem>
+        <OptionsMenuItem onSelect={this.onSelect} isSelected={selectedOption === "groupOption6"} id="groupOption6" key="option 6">Option 2</OptionsMenuItem>
+      </OptionsMenuGroup>
+    ];
+    const toggle = <OptionsMenuToggle onToggle={this.onToggle} toggleTemplate={toggleTemplateText} />
+    
+    return (
+      <OptionsMenu 
+        id="options-menu-align-right-example" 
+        position={OptionsMenuPosition.right} 
+        menuItems={menuGroups} 
+        toggle={toggle} 
+        isOpen={isOpen} 
+        isGrouped />
```

Comment:
if `isGrouped`, the `.pf-c-options-menu__menu` should be a `<div>` instead of a `<ul>`.  The dropdown and app launcher work that way if it helps to use as a reference.

Reply:
hm... the menu is built using the dropdown pieces so i'm surprised that didn't behavior come for free. Nice catch. I'll look into it